uswds/uswds
# USWDS - Bump stylelint, @18f/identity-stylelint-config, and postcss

# Summary

Bump `stylelint`, `@18f/identity-stylelint-config`, and `postcss` to resolve semver vulnerability.

## Breaking change

This is not a breaking change.

## Related issue

Closes Dependabot alert [#61](https://github.com/uswds/uswds/security/dependabot/61)

## Related pull requests

Dependabot Security Update #5366 

## Preview link

Preview link:
USWDS →

## Problem statement

`Stylelint` has vulnerability in semver dependency. 

Updating this dependency alone causes conflicts with `@18f/identity-stylelint-config` which, when updated alone, causes a conflict with `Postcss`

## Solution

Bump `stylelint` and relevant dependencies that are needed in order to prevent errors.

- Bump `stylelint 14.13.0 → 15.10.1`
- Bump `@18f/identity-stylelint-config 1.0.0 → 2.0.0`
- Bump `postcss 8.4.0 → 8.4.19`

## Dependency updates

| Dependency name              | Previous version | New version |
| ---------------------------- | :--------------: | :---------: |
| [stylelint] |     [14.13.0]      |   [15.10.1]   |
| [@18f/identity-stylelint-confige]  |        [ 1.0.0]        |   [2.0.0]   |
| [postcss] |     [8.4.0]     |     [8.4.19]      |

## Testing and Review

1. Pull and checkout this branch
2. Run `npm install` and confirm no installation errors
3. Run `npm run lint:sass` 
4. Confirm it runs without error
5. Confirm no files are changed as a result of linting
6. Run `npm run start`
7. Confirm no build errors
8. Confirm no visual regression

### Testing checklist
- [ ] No install errors
- [ ] Stylelint package dependency resolved
- [ ] Linting runs as expected
- [ ] No visual regression

---
